### PR TITLE
feat(runtime): Add ability to push events to the event queue from WASM

### PR DESCRIPTION
**Pull request by FranklinWaller**
## Motivation

Some tasks require to be executed by a different WASM binary, or in a different VM environment (Like DataRequests, which should have limited capabilities). These decisions on what to do when are decided by the WASM binary, but it needs to be able to trigger events to access these capabilities. This PR introduces trigger_event as a new PromiseAction to enable this.

## Explanation of Changes

Added a trigger_event to the PromiseAction's to enable event pushing to the event queue. I had to include the app actor inside the host actor to let it be able to send messages to the app actor.
At startup the app sends it's own address to the host actor so it can be used later.

## Testing

Run the node normally and trigger the command: `make run-build-all -- -c near cli hello` this will call the node first with the hello world and push an event to the queue (ChainTick). You can see the node executing the VM again with this event.

### CODE REVIEW

File: node/src/host/trigger_event.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/// Communication layer between App & Host adapter
```
Comment (gluax) [inline node/src/host/trigger_event.rs:1]: Is this supposed to be on the struct?

Or file level documentation? If it is file level the doc string should be //! instead of ///

File: node/src/errors.rs [hunk 2]
```diff
@@ -14,6 +14,8 @@ pub enum NodeError {
     RuqliteError(#[from] rusqlite::Error),
     #[error("Chain Adapter Error: {0}")]
     ChainAdapterError(#[from] seda_chains::ChainAdapterError),
+    #[error("Missing app actor address in host adapter, was the node booted?")]
```
Comment (gluax) [inline node/src/errors.rs:17]: Is this error even possible to hit? This is in `Node` so the Node should be booted? Unless it's safety of someone using the library outside of our repo.

File: runtime/core/src/host_adapter.rs [hunk 3]
```diff
@@ -2,7 +2,7 @@ use std::fmt::Display;
 
 use seda_chains::Client;
 use seda_config::{ChainConfigs, NodeConfig};
-use seda_runtime_sdk::Chain;
+use seda_runtime_sdk::{events::Event, Chain};
 
 #[async_trait::async_trait]
 pub trait HostAdapter: Send + Sync + Unpin + 'static {
```
Comment (gluax) [inline runtime/core/src/host_adapter.rs:8]: I question the need for this to still be a trait. But even if we change that it should be in another PR.

Comment (FranklinWaller) [reply runtime/core/src/host_adapter.rs:8]: Yeah agree, let's discuss this later

File: node/src/errors.rs (hunk 2 above)
Comment (FranklinWaller) [reply node/src/errors.rs:17]: It probably won't hit, but since we are avoiding unwraps, and in the rare case it might happen, there's at least something for it

**Review by mariocao — APPROVED** [review]

**Review by gluax — APPROVED** [review]
